app/drivers/usb/hid.c

```diff
@@ -151,6 +160,16 @@ static uint8_t hid_setup(USBD_HandleTypeDef *p_dev, USBD_SetupReqTypedef *p_req)
 	return HAL_OK;
 }
 
+static uint8_t hid_ep0_rx_ready(USBD_HandleTypeDef *p_dev)
+{
+	if (!self.report_available)
+		return HAL_OK;
+
+	self.report_available = false;
```

Comment:
This can probably be changed to xSemaphoreGiveFromISR later on i guess

Reply:
Yes, that's the idea